File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-6.cpp
class Solution {
public:
    bool mergeTriplets(vector<vector<int>>& triplets, vector<int>& target) {
        bool x{}, y{}, z{};
        for (const auto& triplet : triplets) {
            if (triplet[0] > target[0] || triplet[1] > target[1] || triplet[2] > target[2]) 
                continue;
            if (triplet[0] == target[0]) x = true;
            if (triplet[1] == target[1]) y = true;
            if (triplet[2] == target[2]) z = true;
        }
        return x && y && z;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-6.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int num = nums[i];
            if (m.containsKey(target - num)) {
                return new int[]{m.get(target - num), i};
            }
            m.put(num, i);
        }
        return new int[]{-1, -1};
    }
}
